### PR TITLE
Run static type checking with Mypy

### LINKED ISSUE #114: Publish with `py.typed`
https://peps.python.org/pep-0561/

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes #114

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
`py.typed` is included in the distribution: https://github.com/pytest-dev/pytest-subtests/actions/runs/6951118504?pr=115#summary-18912531061

File: tox.ini [hunk 1]
```diff
@@ -12,3 +12,7 @@ deps =
 
 commands =
     pytest {posargs:tests}
+
+[testenv:mypy]
+deps = mypy
```
Comment (RonnyPfannschmidt) [inline tox.ini:17]:
```suggestion
deps = 
  mypy
  pytest
  
```

File: pyproject.toml [hunk 2]
```diff
@@ -5,4 +5,8 @@ requires = [
 ]
 build-backend = "setuptools.build_meta"
 
+[tool.mypy]
```
Comment (RonnyPfannschmidt) [inline pyproject.toml:8]: to be properly usable for externals we need to be closer to strict than the defaults

i haven't yet take an look how much pain is in that change but i believe we need to disallow untyped defs

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]
thanks for getting this started!

File: tox.ini (hunk 1 above)
Comment (edgarrmondragon) [reply tox.ini]: 91b5f1a

File: pyproject.toml (hunk 2 above)
Comment (edgarrmondragon) [reply pyproject.toml:8]: 91b5f1ad40d0d3e4d818ae9377c608cb766cbdbd

**Comment by edgarrmondragon** [comment]
Thanks for the feedback @RonnyPfannschmidt! Please take a look at your convenience.

File: setup.cfg [hunk 3]
```diff
@@ -23,20 +23,27 @@ classifiers =
...
     Operating System :: OS Independent
     License :: OSI Approved :: MIT License
+    Typing :: Typed
 keywords = test, unittest, pytest
 
 [options]
-py_modules = pytest_subtests
 install_requires =
     pytest>=7.0
     attrs>=19.2.0
```
Comment (nicoddemus) [inline setup.cfg:32]:
```suggestion
    attrs>=19.2.0
    pytest>=7.0
    typing_extensions;python_version<"3.8"
```

**Review by nicoddemus — APPROVED** [review]
Great work, thanks!

**Comment by nicoddemus** [comment]
(Working on the mypy failures)

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -15,6 +15,13 @@ repos:
     rev: v3.12.0
     hooks:
     -   id: reorder-python-imports
+-   repo: https://github.com/pre-commit/mirrors-mypy
```
Comment (nicoddemus) [inline .pre-commit-config.yaml:18]: FYI we in general rely on running mypy with pre-commit instead of having a dedicated job on CI. 👍

**Comment by nicoddemus** [comment]
Thanks again @edgarrmondragon! 

Wait for @RonnyPfannschmidt's review before merging. 👍

**Review by RonnyPfannschmidt — APPROVED** [review]

**Pull request merged**